src/reverse_complement.rs

```diff
@@ -10,12 +10,11 @@ extern crate rayon;
 extern crate memchr;
 
 use std::io::{Read, Write};
-use std::{io, ptr, slice};
+use std::{cmp, io, mem};
 use std::fs::File;
 
 struct Tables {
```

Comment:
What do you think about removing this `struct` and just having a `table: &[u8; 256]` passed around and a

```rust
fn build_table() -> [u8; 256] {
    let mut table = [0u8; 256];
    for (i, v) in table8.iter_mut().enumerate() {
        *v = match i as u8 {
             b'A' | b'a' => b'T',
             b'C' | b'c' => b'G',
             b'G' | b'g' => b'C',
             b'T' | b't' => b'A',
             b'U' | b'u' => b'A',
             b'M' | b'm' => b'K',
             b'R' | b'r' => b'Y',
             b'W' | b'w' => b'W',
             b'S' | b's' => b'S',
             b'Y' | b'y' => b'R',
             b'K' | b'k' => b'M',
             b'V' | b'v' => b'B',
             b'H' | b'h' => b'D',
             b'D' | b'd' => b'H',
             b'B' | b'b' => b'V',
             b'N' | b'n' => b'N',
             i => i,
         };
    }
}
```

called once in `main`. Or we could even go one better and do what C does:

```rust
const TABLE: &'static [u8] =
    b"                                                                 \
       TVGH  CD  M KN   YSAABW R       TVGH  CD  M KN   YSAABW R";
```